docs/rfc/rfc-9.adoc

```diff
@@ -0,0 +1,145 @@
+:toc: macro
+
+= RFC 9: Bitcoin Ordinals
+
+:icons: font
+:numbered:
+toc::[]
+
+== Background
+
+As of 2023-02-23, Bitcoin Ordinals are 🔥 HOT 🔥. Here is a
+https://insights.glassnode.com/ordinal-theory-and-the-rise-of-inscriptions/[decent
+explainer].
+
+image:assets/inscriptions-over-time.png[]
+
+There is not a good way to get those Ordinals over to Ethereum currently, where
+Ordinal owners can use the more mature NFT platforms. We could be that way!
+
+Here is an https://ordinals.com/inscription/cfa453dbf693641b8f0c6b83cb733e71832606fc26415c36c70744953d6f6161i0[example ordinal].
+
+=== Current Functionality
+
+Ordinals, at a high level, are single satoshis with metadata stuck into their
+witness field (called Inscribing). Each satoshi has an immutable age, which
+effectively gives them serial numbers.
+
+These Inscribed Satoshis (called Inscriptions) can be transferred like normal
+BTC. The sender's wallet software needs to be careful to not merge the
+Inscriptions with "normal" UTXOs, and needs to be careful to not accidentally
+send the Inscription as part of a payment. The
+https://github.com/casey/ord[casey/ord] repo provides the primatives for the
+handling of Ordinals on BTC.
+
+A good mental model here is thinking about Inscriptions like
+https://www.apmex.com/product/117942/1932-s-washington-quarter-choice-au[special
+edition quarters]. The the linked quarter trades for ~$350, but is also legal
+tender worth $0.25. An owner of such a quarter probably does not want to use it
+to pay for parking. Inscriptions work the same way.
+
+Since Ordinals are still BTC, we can use most of the same tBTCv2 ideas for
+custodying Ordinals: rotating wallets, P2SH addresses, fraud proofs,
+redemptions, etc, with some slight modifications.
+
+== Proposal
+
+=== Overview
+
+We build a BTC <> Ethereum Ordinal bridge!
+
+We create 51-of-100 threshold signing groups using the same process that we use
```

Comment:
Are they the same wallets as the current instance of the bridge, or are they separate groups?

Reply:
I was imagining while writing this that they were totally separate, but I think they can be in the same wallet without much risk. The v2 system is configured to only sweep things based on particular ethereum events, and only redeem from the main UTXO.

Deposited Ordinals would never trigger that same event, would never be swept into the main UTXO, and so we would never have a chance of accidentally sending them to folks. We'd save gas on wallet creation, which is nice. 

A tricky bit is wallet closures. We'd have two default conditions

1. when it has too little BTC 
2. when it has too few Ordinals

And could either close a wallet when *either* conditions are true (more wallet closures) or when *both* conditions are true (less wallet closures)

Reply:
[184bc8](https://github.com/keep-network/tbtc-v2/pull/561/commits/184bc81094bdb3e10ff9bad7f77cbcad13b496a3)

Reply:
I'm totally down to have the wallets be separate

Reply:
I read through [send.rs](https://github.com/casey/ord/blob/master/src/subcommand/wallet/send.rs), [transaction_builder.rs](https://github.com/casey/ord/blob/master/src/subcommand/wallet/transaction_builder.rs) and [index.rs](https://github.com/casey/ord/blob/master/src/index.rs)

It seems like `ord` needs access to a full node because it needs to load up the whole history into its special format. When you want to send an ordinal, it tries to call `get_inscription_satpoint_by_id`, which means that its database needs to know where that particular inscription ID is located, and if we're moving it using a special P2SH script, it won't.

That isn't a dealbreaker, it just means that we need to extend or modify how the ord client works if we want to support refunds. I *think* we'd also have to do the same thing if we wanted to support batch transactions.